src/Engine/ProtoCore/Reflection/MemberInfoReflectionExtensions.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Autodesk.DesignScript.Runtime;
+
+namespace ProtoFFI.Reflection
+{
+    internal static class MemberInfoReflectionExtensions
+    {
+        internal static MethodInfo GetBaseDefinitionReflectionContext(this MethodInfo methodInfo)
+        {
+            try
+            {
+                // if overriding object 
+                var baseType = methodInfo.DeclaringType.BaseType;
+                if (baseType is null)
+                {
+                    return methodInfo;
+                }
+
+                // This will fail if there are overloads of the method
+                var baseTypeMethod = baseType.GetMethod(methodInfo.Name);
```

Comment:
what about getting all overloads and then searching for the overload with the correct number of parameters, you should have the parameter names as well which you can use to pick the correct overload - is that correct?

Reply:
yea this was the approach i was trying to go after, problem is that in order to get the parameters from the `MethodInfo` we need to use `GetParameters()` which will fail if the parameter type is not loaded. This seems to be the only way of retrieving parameters from the method

Reply:
right - and if the type is not loaded - what about using https://docs.microsoft.com/en-us/dotnet/api/system.type.reflectiononlygettype?view=net-5.0 to load it directly ?

Reply:
it is a bummer we can't get the parameter names without the types - seems like there is some way around it - but in any case, it reminds me that this tool should have a known issues list - and this can be one of them :) 

maybe we can put a readme in this folder if there isn't already one?

Reply:
> right - and if the type is not loaded - what about using https://docs.microsoft.com/en-us/dotnet/api/system.type.reflectiononlygettype?view=net-5.0 to load it directly ?

That could probably work since we should have the references in the PathResolver now, maybe there are edge case where it dosent have the needed reference, ill do some testing and make some notes if there are scenarios were it wont work.

Reply:
this got my thinking that for some of the complexity you are running into where reflection only contexts don't provide the same tools as regular contexts you might want to try Cecil - 

https://github.com/jbevain/cecil

it's claim to fame is generating new patched assemblies - but it can also inspect .net il code without using reflection...